demo/ota_demo.c

```diff
@@ -86,22 +87,31 @@ bool otaDemo_handleIncomingMQTTMessage( char * topic,
     return handled;
 }
 
-/* TODO: Implement for the Jobs library */
-static bool handleJobsStartNextAcceptedCallback( const char * jobId,
-                                                 const size_t jobIdLength,
-                                                 const char * jobDoc,
-                                                 const size_t jobDocLength )
+static bool jobHandlerChain( char * message, size_t messageLength )
 {
+    char * jobDoc;
+    size_t jobDocLength = 0U;
+    char * jobId;
+    size_t jobIdLength = 0U;
     bool handled = false;
+
     if( globalJobId[ 0 ] == 0 )
     {
         strncpy( globalJobId, jobId, jobIdLength );
+    }
+
+    jobDocLength = coreJobs_getJobDocument( message, messageLength, &jobDoc );
+    jobIdLength = coreJobs_getJobId( message, messageLength, &jobId );
```

Comment:
I think we should strncpy the jobID after coreJobs_getJobID has been called.

Reply:
That's a wise observation...what a slip up on my part haha